beacon_node/beacon_chain/src/test_utils.rs

```diff
@@ -450,13 +455,6 @@ where
         honest_fork_blocks: usize,
         faulty_fork_blocks: usize,
     ) -> (Hash256, Hash256) {
-        let initial_head_slot = self
```

Comment:
A lot of other tests that use this function expect the original behaviour of forking from the `initial_head_slot`, so I think we should leave this as it was. The reason you were encountering finalized block errors was that your test provided 2/3 of validators to the honest chain (which gets created first, and gets finalized), and then tries to create a faulty fork before that finalization point. If you provide less than 2/3 honest validators then the honest fork won't immediately finalize and creating the faulty fork won't trigger any errors. That's what the other tests do.

I agree that this primitive for creating forks isn't the most intuitive or flexible, and that we should rethink how these tests are structured – I'd be interested to talk to you about your ideas for restructuring sometime!

Reply:
Seems to me like I was just testing a different scenario that's also a valid one and valuable.  I've restored the original code and put my scenario in a separate function (`fork_chain`).